src/utils/dayViewLayout.js

```diff
@@ -355,10 +358,56 @@ export function getStyledAvailabilities ({
     totalMin,
   };
 
-  availabilities.forEach((availability, availabilityIdx) => {
-    let style = getYStyles(availabilityIdx, helperArgs);
-    styledAvailabilities.push({ availability, style });
-  })
+  const overlapMap = {};
+  let columnIndex = 0;
+
+  for (let availabilityIndex = 0; availabilityIndex < availabilities.length; availabilityIndex++) {
+    const availability = availabilities[availabilityIndex];
+    let grouped = false;
+
+    // Check for overlap with existing groups
+    const overlapKeys = Object.keys(overlapMap);
+    for (let groupIndex = 0; groupIndex < overlapKeys.length; groupIndex++) {
+      const groupKey = overlapKeys[groupIndex];
+      const group = overlapMap[groupKey];
+      const lastAvailability = group[group.length - 1];
+      const lastAvailabilityEndTime = get(lastAvailability, availabilityEndAccessor);
+      const availabilityStartTime = get(availability, availabilityStartAccessor);
+      if (lastAvailabilityEndTime <= availabilityStartTime) {
+        // No overlap, add to current group
+        group.push(availability);
+        grouped = true;
+        break;
+      }
+    }
+
+    if (!grouped) {
+      // Create a new group
+      overlapMap[columnIndex] = [availability];
+      columnIndex++;
+    }
+  }
+
+  const overlapKeys = Object.keys(overlapMap);
+  for (let groupIndex = 0; groupIndex < overlapKeys.length; groupIndex++) {
```

Comment:
Can we use `Array.prototype.forEach` instead? I don't see any `break` statements for this for-loop, and unless a break statement is needed, `forEach` is pretty much always more readable.